### PR TITLE
Merge Alpine linux x86 support

### PR DESCRIPTION
**Pull request by superajun-wsj**

### CODE REVIEW

**Review by yuleil — COMMENTED** [review]
Two questions:

1. Will there be any impact on the build of the glibc version after the merger?
2. Does deleting thread_db affect users?

File: src/hotspot/os_cpu/linux_aarch64/os_linux_aarch64.cpp [hunk 1]
```diff
@@ -74,7 +74,6 @@
 # include <pwd.h>
 # include <poll.h>
 # include <ucontext.h>
-# include <fpu_control.h>
```
[the remark this answers is not in the document]
Comment (superajun-wsj) [reply src/hotspot/os_cpu/linux_aarch64/os_linux_aarch64.cpp:77]: fpu_control.h is not needed in this file, it seems only needed by x86 and arm, while not needed x86_64 or aarch64. this change is also from upstream: https://github.com/openjdk/jdk/commit/63009f90

File: src/hotspot/os/linux/os_linux.cpp [hunk 2]
```diff
@@ -832,7 +849,17 @@ bool os::create_thread(Thread* thread, ThreadType thr_type,
   assert(is_aligned(stack_size, os::vm_page_size()), "stack_size not aligned");
 
   int status = pthread_attr_setstacksize(&attr, stack_size);
-  assert_status(status == 0, status, "pthread_attr_setstacksize");
+  if (status != 0) {
```
[the remark this answers is not in the document]
Comment (superajun-wsj) [reply src/hotspot/os/linux/os_linux.cpp:852]: it's from an upstream commit: https://bugs.openjdk.java.net/browse/JDK-8241053,  it's compatible with both Alpine or linux environment like this.

**Comment by superajun-wsj** [comment]
1. Will there be any impact on the build of the glibc version after the merger?
2. Does deleting thread_db affect users?
@yuleil , 1. this pr will not any impact on the build of glibc after the merge, the patch has passed both x86-64 and aarch64 CI pipeline. 2. deleting thread_db will not affect users, because the commit(https://bugs.openjdk.java.net/browse/JDK-8181313) used another way to realize the same function it acts.

**Review by yuleil — APPROVED** [review]
LGTM

**Review by kuaiwei — APPROVED** [review]

**Pull request merged**